### PR TITLE
1861 check project status

### LINKED ISSUE #1861: API checks if a project has been initialised
**User Story**
As a user of kubermatic API server I'd like it to detect and signal back when I'm trying to do operations associated with projects resource that hasn't been initialised yet. Creating a project is a asynchronous operation. It kicks off the controller that performs initialisation. The controller updates the status of the project once it finishes. API server should check if the project resource has been initialised and if not it should yield back an error. Possibles error codes could be 202 or 503 or something else ? (See also: https://app.zenhub.com/workspace/o/kubermatic/kubermatic/issues/1860)

How to check if a project has been initialised ?
```
type Project struct {
	NewObjectMeta
	Status string `json:"status"`
}
```
the project's status field must be set to `active`

Possible implementation:
Change GET method of the rbac compliant project provider so that it accepts "options" (similarly to List method). If the option is not set then the provider will not check if the project was initialised. By default (if the option was not set) it checks if the project has been initialised. 

Next identify all possible places that don't require a project to be initialised in our API server and set appropriate option when getting the project.

**Acceptance criteria**
- API verifies if the given project has been initialised before allowing to an operation that requires it.

### PR DESCRIPTION
**Pull request by glower**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1861

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by glower** [comment]
added 3 lines of code, then fixing tests for the whole day

File: api/pkg/provider/types.go [hunk 1]
```diff
@@ -108,6 +108,13 @@ type ClusterGetOptions struct {
 	CheckInitStatus bool
 }
 
+// ProjectGetOptions allows to check the status of the Project
+type ProjectGetOptions struct {
+	// SkipInitStatus if set to true will skip the check if project is initialized. By defoult the call will return
+	// an  error if not all project components are running
+	SkipInitStatus bool
```
Comment (mrIncompetent) [inline api/pkg/provider/types.go]: Can we maybe use `IncludeUninitialized bool` ?
That would be the same option like Kubernetes uses it.

Comment (glower) [reply api/pkg/provider/types.go]: Done

File: api/pkg/provider/kubernetes/project.go [hunk 2]
```diff
@@ -117,15 +118,22 @@ func (p *ProjectProvider) Delete(user *kubermaticapiv1.User, projectInternalName
...
 	if err != nil {
 		return nil, err
 	}
-	return masterImpersonatedClient.Projects().Get(projectInternalName, metav1.GetOptions{})
+	project, err := masterImpersonatedClient.Projects().Get(projectInternalName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if !options.IncludeUninitialized && project.Status.Phase != kubermaticapiv1.ProjectActive {
+		return nil, kerrors.NewServiceUnavailable("Project components are not ready yet")
```
Comment (mrIncompetent) [inline api/pkg/provider/kubernetes/project.go]: Maybe the error message should be changed to `Project is not initialized yet` ?
Or are we waiting for project components here?

**Comment by glower** [comment]
error message changed

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**